ticketutil/bugzilla.py

```diff
@@ -114,62 +114,76 @@ def _verify_project(self, project):
+    def get_ticket_content(self, ticket_id=None):
         """
-        Queries the Bugzilla API to see if ticket_id is a valid ticket for the given Bugzilla instance.
-        :param ticket_id: The ticket you're verifying.
-        :return: True or False depending on if ticket is valid.
+        Queries the Bugzilla API to get ticket_content using ticket_id.
+        :param ticket_id: ticket number, if not set self.ticket_id is used.
+        :return: self.request_result: Named tuple containing request status, error_message, and url info.
         """
+        if ticket_id is None:
+            ticket_id = self.ticket_id
+            if not self.ticket_id:
+                error_message = "No ticket ID associated with ticket object. " \
+                                "Set ticket ID with set_ticket_id(<ticket_id>)"
+                logger.error(error_message)
+                return self.request_result._replace(status='Failure', error_message=error_message)
         try:
             r = self.s.get("{0}/{1}".format(self.rest_url, ticket_id))
-            logger.debug("Verify ticket_id: status code: {0}".format(r.status_code))
+            logger.debug("Get ticket content: status code: {0}".format(r.status_code))
             r.raise_for_status()
+            self.ticket_content = r.json()
+            return self.request_result._replace(ticket_content=self.ticket_content)
         except requests.RequestException as e:
-            logger.error("Unexpected error occurred when verifying ticket_id")
+            error_message = "Error getting ticket content"
+            logger.error(error_message)
             logger.error(e)
-            return False
-
-        error_responses = ["Bug #{0} does not exist.".format(ticket_id),
-                           "\\\"{0}\\\" is out of range for type integer".format(ticket_id),
-                           "\'{0}\' is not a valid bug number nor an alias to a bug.".format(ticket_id)]
+            return self.request_result._replace(status='Failure', error_message=error_message)
 
-        # Bugzilla's API returns 200 even if the request response is not valid. We need to parse r.text.
-        if any(error in r.text for error in error_responses):
+    def _verify_ticket_id(self, ticket_id):
+        """
+        Check if ticket_id is connected with valid ticket for the given Bugzilla instance.
+        :param ticket_id: The ticket you're verifying.
+        :return: True or False depending on if ticket is valid.
+        """
+        self.request_result = self.get_ticket_content(ticket_id)
+        if 'Failure' in self.request_result.status:
             logger.error("Ticket {0} is not valid".format(ticket_id))
             return False
         else:
             logger.debug("Ticket {0} is valid".format(ticket_id))
             return True
 
-    def create(self, summary, description, **kwargs):
+    def create(self, summary, description, component, version, **kwargs):
```

Comment:
So, I'm not sure about making component and version required fields for create(). Sure, they may be required for our Red Hat Bugzilla instance, but this may just be a setting that our Bugzilla admins decided on. We want to make sure ticketutil works with any Bugzilla instance out there. Have you tested any other instances to see if they also require component and version when creating a ticket?

Other ways to research without testing on a second Bugzilla instance are to research the API docs or see what other python bugzilla modules use as required fields.

Reply:
All these parameters are required, but some can be set default. Check http://bugzilla.readthedocs.io/en/latest/api/core/v1/bug.html#create-bug

Reply:
Oh, yep. Thanks for looking that up. :)